crates/sui-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -171,19 +171,40 @@ where
                 },
             );
 
-        // TODO: Use multi get objects
         let rpc = self.rpc_client.clone();
-        let all_mutated_objects = join_all(all_mutated.into_iter().map(|(id, version, status)| {
-            rpc.read_api()
-                .try_get_parsed_past_object(id, version, SuiObjectDataOptions::bcs_lossless())
-                .map(move |resp| (resp, status))
-        }))
-        .await
-        .into_iter()
-        .try_fold(vec![], |mut acc, (response, status)| {
-            acc.push((status, response?.into_object()?));
-            Ok::<_, Error>(acc)
-        })?;
+        let all_mutated_objects =
+            join_all(all_mutated.chunks(MULTI_GET_CHUNK_SIZE).map(|objects| {
+                let wanted_past_object_statuses: Vec<ObjectStatus> =
+                    objects.iter().map(|(_, _, status)| *status).collect();
+
```

Comment:
not related to your PR, but I think we did not fetch `deleted` object above and thus the object table will be outdated? cc @gegaowp @patrickkuo

Reply:
this is past objs and `all_deleted_objects` are handled indeed on `objects` table, so seems fine to me.

Reply:
for objects table, there is a separate
```
                TransactionObjectChanges {
                    mutated_objects: all_mutated_objects,
                    deleted_objects: all_deleted_objects,
                }
```